src/NodeVisitor/OptimizerNodeVisitor.php

```diff
@@ -42,6 +42,7 @@ final class OptimizerNodeVisitor implements NodeVisitorInterface
 {
     public const OPTIMIZE_ALL = -1;
     public const OPTIMIZE_NONE = 0;
+    public const OPTIMIZE_PRINT = 1;
```

Comment:
making this optimization configurable would deserve a dedicated PR (as it deserve a changelog entry for such new feature)

Reply:
Makes sense, i extracted those changes into a separate PR https://github.com/twigphp/Twig/pull/4051.